packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolCreateEntity.ts

```diff
@@ -0,0 +1,115 @@
+import { useLazyQuery, useMutation } from "@apollo/client";
+import { useCallback } from "react";
+
+import {
+  GetEntityTypeRootedSubgraphQuery,
+  GetEntityTypeRootedSubgraphQueryVariables,
+  CreatePersistedEntityMutation,
+  CreatePersistedEntityMutationVariables,
+} from "../../../../graphql/apiTypes.gen";
+import { getEntityTypeRootedSubgraphQuery } from "../../../../graphql/queries/ontology/entity-type.queries";
+import { createPersistedEntityMutation } from "../../../../graphql/queries/knowledge/entity.queries";
+import { CreateEntityMessageCallback } from "./knowledge-shim";
+
+export const useBlockProtocolCreateEntity = (
+  readonly?: boolean,
+): {
+  createEntity: CreateEntityMessageCallback;
+} => {
+  const [createFn] = useMutation<
+    CreatePersistedEntityMutation,
+    CreatePersistedEntityMutationVariables
+  >(createPersistedEntityMutation);
+
+  const [getEntityTypeRootedSubgraphFn] = useLazyQuery<
+    GetEntityTypeRootedSubgraphQuery,
+    GetEntityTypeRootedSubgraphQueryVariables
+  >(getEntityTypeRootedSubgraphQuery);
+
+  const createPersistedEntity: CreateEntityMessageCallback = useCallback(
+    async ({ data }) => {
+      if (readonly) {
+        return {
+          errors: [
+            {
+              code: "FORBIDDEN",
+              message: "Operation can't be carried out in readonly mode",
+            },
+          ],
+        };
+      }
+
+      if (!data) {
+        return {
+          errors: [
+            {
+              code: "INVALID_INPUT",
+              message: "'data' must be provided for createPersistedEntity",
+            },
+          ],
+        };
+      }
+
+      const { entityTypeId, properties } = data;
+
+      const { data: responseData } = await createFn({
+        variables: {
+          entityTypeId,
+          properties,
+        },
+      });
+
+      const { createPersistedEntity: createdEntity } = responseData ?? {
+        createPersistedEntity: null,
```

Comment:
I suppose this suggestion is similar to what we did for `updateEntity`, could we have empty braces here instead? 
like in https://github.com/hashintel/hash/blob/4f14a1ec27bade45b62994508d86ab71bdb32ef8/packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolUpdateEntity.ts#L61

Reply:
Thanks for spotting, fixed in https://github.com/hashintel/hash/pull/1178/commits/cf32e2e80595952918fd0e4d1609549d30d8f8da